types/metadataV14.go

```diff
@@ -0,0 +1,419 @@
+package types
+
+import (
+	"fmt"
+	"hash"
+	"strings"
+
+	"github.com/centrifuge/go-substrate-rpc-client/v3/scale"
+	"github.com/centrifuge/go-substrate-rpc-client/v3/xxhash"
+)
+
+// Based on:
+// https://tinyurl.com/metadataV14
+type MetadataV14 struct {
+	Lookup    PortableRegistryV14
+	Pallets   []PalletMetadataV14
+	Extrinsic ExtrinsicV14
+	Type      Si1LookupTypeID
+
+	// Custom field to help us lookup a type from the registry
+	// more efficiently. This field is built while decoding and
+	// it is not to be encoded.
+	EfficientLookup map[int64]*Si1Type
```

Comment:
I think we can use tag annotations like `scale:"-"` to tell the scale encoder/decoder to omit this field. Then we can simplify the code below, can you check that it works, this is the code that checks the tag option: https://github.com/centrifuge/go-substrate-rpc-client/blob/c28fe362a18bbe712ce445b0d6fd27f048d63485/scale/codec.go#L229

Reply:
TIL! Thanks 👍